lib/Doctrine/DBAL/Platforms/MariaDb102Platform.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\DBAL\Platforms;
+
+use Doctrine\DBAL\Types\Type;
+
+/**
+ * Provides the behavior, features and SQL dialect of the MariaDB 10.2 (10.2.7 GA) database platform.
+ *
+ * Note: Should not be used with versions prior ro 10.2.7.
+ *
+ * @author Vanvelthem Sébastien
+ * @link   www.doctrine-project.org
+ */
+final class MariaDb102Platform extends MySqlPlatform
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function hasNativeJsonType() : bool
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     * @link https://mariadb.com/kb/en/library/json-data-type/
+     */
+    public function getJsonTypeDeclarationSQL(array $field) : string
+    {
+        return 'JSON';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getReservedKeywordsClass() : string
+    {
+        return Keywords\MariaDb102Keywords::class;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function initializeDoctrineTypeMappings() : void
+    {
+        parent::initializeDoctrineTypeMappings();
+
+        $this->doctrineTypeMapping['json'] = Type::JSON;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * Since MariaDB 10.2.1 blob and text columns can have a default value.
+     * @link https://mariadb.com/kb/en/library/blob-and-text-data-types/
+     */
+    protected function isDefaultValueSupportedForType(Type $field) : bool
+    {
+        return true;
```

Comment:
That's a bold statement :-) Does MariaDB support default values for _all_ types?

Reply:
Except JSON: https://dev.mysql.com/doc/refman/5.7/en/data-type-defaults.html. But let's assume JSON and BLOB/TEXT are the same (they are aliased in maria - mysql stores json in binary BLOB). Cannot find on mariadb doc (403 forbidden).

Let's first decide if I include default values